src/core/unit.c

```diff
@@ -2944,7 +2944,7 @@ void unit_unwatch_pidref(Unit *u, PidRef *pid) {
 
         if (uu == u)
                 /* OK, we are in the first table. Let's remove it there then, and we are done already. */
-                assert_se(hashmap_remove_value(u->manager->watch_pids, pid2, uu) == uu);
+                assert_se(hashmap_remove_value(u->manager->watch_pids, pid2, uu));
```

Comment:
hmm, why drop this? this is a safety check that the entry matches what we expect here.

Reply:
ah, i see why that's unnecessary.

Reply:
Please check the commit message - the only difference between `hashmap_remove` and `_remove_value` is that the latter does the check internally, and returns NULL if things don't match IIUC?

Reply:
It always return `uu` or `NULL`, and `NULL` is avoided by the assertion. So, I think this change is reasonable.